### DISCUSSION TITLE
How to easily add #+BEGIN_SRC blocks?

### DISCUSSION THREAD

**Discussion by Maltimore** [nvim-orgmode/orgmode#379] [Q&A] [answered]
Hi,
this is a super simple question. I don't have any plugins installed that input entire snippets, so I'm wondering how you easily add these blocks:
```
#+BEGIN_SRC
some code here
#+END_SRC
```
If there's no easy way to do it without a snippet plugin, I'll happily take suggestion for a plugin.

**Comment by refaelsh** [comment]
Please watch this excellent video: https://www.youtube.com/watch?v=XA2WjJbmmoM.
Somewhere towards the end, he explains how to do snippets without plugins.
If you want a plugin, I recommend this one: https://github.com/hrsh7th/nvim-cmp from here: https://github.com/nvim-orgmode/orgmode#completion.

**Reply by Maltimore** [reply]
Thanks, that video was refreshing (I only watched the snippets part).
Yep, I've seen the nvim-cmp plugin as well, but it seems really complex to set up and you somehow need many different plugins to make it work it seems? I'm more looking for something (more) minimalistic.

**Reply by refaelsh** [reply]
I don't know any other snipets plugins to recommend. I my self use the method from the video. But I do use the `nvim-cmp` for the completions it provides.

**Comment by jgollenz** [comment, marked as answer]
I'm (kinda) using [LuaSnip](https://github.com/L3MON4D3/LuaSnip) for things like that, which seems to be the most popular snippet engine for neovim

**Reply by Maltimore** [reply]
That looks pretty good, I'll check that out next. Thanks!

**Reply by Maltimore** [reply]
I'll mark it as solved because my main question was whether there's a vim-orgmode specific way to add such code blocks.

**Reply by jgollenz** [reply]
There is at least auto-completion for `#+BEGIN_SRC` and `#+END_SRC`, but not for automatically adding the end tag when you insert the begin tag (at least not that I'm aware of).
